src/Content/OEmbed.php

```diff
@@ -48,132 +47,128 @@ public static function replaceCallback($matches)
 	/**
 	 * @brief Get data from an URL to embed its content.
 	 *
-	 * @param string $embedurl The URL from which the data should be fetched.
-	 * @param bool $no_rich_type If set to true rich type content won't be fetched.
+	 * @param string $embedurl     The URL from which the data should be fetched.
+	 * @param bool   $no_rich_type If set to true rich type content won't be fetched.
 	 *
-	 * @return bool|object Returns object with embed content or false if no embeddable
-	 * 	                   content exists
+	 * @return \Friendica\Object\OEmbed
 	 */
 	public static function fetchURL($embedurl, $no_rich_type = false)
 	{
-		$embedurl = trim($embedurl, "'");
-		$embedurl = trim($embedurl, '"');
+		$embedurl = trim($embedurl, '\'"');
 
 		$a = get_app();
 
+		$cache_key = 'oembed:' . $a->videowidth . ':' . $embedurl;
+
 		$condition = ['url' => normalise_link($embedurl), 'maxwidth' => $a->videowidth];
-		$oembed = DBA::selectFirst('oembed', ['content'], $condition);
-		if (DBA::isResult($oembed)) {
-			$txt = $oembed["content"];
+		$oembed_record = DBA::selectFirst('oembed', ['content'], $condition);
+		if (DBA::isResult($oembed_record)) {
+			$json_string = $oembed_record['content'];
 		} else {
-			$txt = Cache::get($a->videowidth . $embedurl);
+			$json_string = Cache::get($cache_key);
 		}
+
 		// These media files should now be caught in bbcode.php
 		// left here as a fallback in case this is called from another source
-
-		$noexts = ["mp3", "mp4", "ogg", "ogv", "oga", "ogm", "webm"];
+		$noexts = ['mp3', 'mp4', 'ogg', 'ogv', 'oga', 'ogm', 'webm'];
 		$ext = pathinfo(strtolower($embedurl), PATHINFO_EXTENSION);
 
+		$oembed = new \Friendica\Object\OEmbed($embedurl);
 
-		if (is_null($txt)) {
-			$txt = "";
+		if ($json_string) {
```

Comment:
Will this work even when ```$json_string``` is ```null```?

Reply:
Yes, as long as it's been declared previously.